File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02917/p02917_num8_parsed.py
def find_maximum_sum(N, B):
    A = [0] * (N+1)
    A[0] = B[0]
    A[N] = B[N-2]
    for i in range(1, N):
        A[i] = min(B[i-1], B[i])
    return sum(A)

# Read input from standard input
N = int(input())
B = list(map(int, input().split()))

# Call the function
result = find_maximum_sum(N, B)

# Print the result
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02917/p02917_num8_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    // Read the value of N
    int N;
    cin >> N;
    
    // Read the sequence B
    vector<int> B(N-1);
    for (int i = 0; i < N-1; i++) {
        cin >> B[i];
    }
    
    // Calculate the maximum possible sum of the elements of A
    int sum_A = B[0] + B[N-2];
    for (int i = 1; i < N-1; i++) {
        sum_A += min(B[i-1], B[i]);
    }
    
    // Print the result
    cout << sum_A << endl;
    
    return 0;
}